sound/hda/intel-nhlt.c

```diff
@@ -298,7 +302,21 @@ intel_nhlt_get_endpoint_blob(struct device *dev, struct nhlt_acpi_table *nhlt,
 	dev_dbg(dev, "Looking for configuration:\n");
 	dev_dbg(dev, "  vbus_id=%d link_type=%d dir=%d, dev_type=%d\n",
 		bus_id, link_type, dir, dev_type);
-	dev_dbg(dev, "  ch=%d fmt=%d/%d rate=%d\n", num_ch, vbps, bps, rate);
+	if (link_type == NHLT_LINK_DMIC && bps == 32) {
+		/*
+		 * Ignore the vbps matching for 32 bps for DMIC since some NHLT
+		 * table have the vbps as 24, some have it as 32.
+		 * The DMIC hardware supports only one type of 32 bit sample
+		 * size, which is 24 bit sampling on the MSB side and bits[1:0]
+		 * is used for indicating the channel number.
+		 */
+		dev_dbg(dev,
+			"  ch=%d fmt=%d rate=%d (vbps is ignored for DMIC 32bit format)\n",
+			num_ch, bps, rate);
+		vbps = 0;
+	} else {
+		dev_dbg(dev, "  ch=%d fmt=%d/%d rate=%d\n", num_ch, vbps, bps, rate);
+	}
```

Comment:
I would move this to the SOF driver.

Reply:
not following @ujfalusi, the caller can set vbs to zero and ask the intel-nhlt layer to give the first blob that matches bps.

Reply:
@plbossart, right, I have overlooked that. Do we really want that? I'm not sure.